pyccl/tests/test_pk2d.py

```diff
@@ -65,6 +66,60 @@ def test_pk2d_smoke():
     assert_(not np.isnan(psp.eval(1E-2, 0.5, cosmo)))
 
 
+@pytest.mark.parametrize('model', ['bbks', 'eisenstein_hu'])
+def test_pk2d_from_model(model):
+    cosmo = ccl.Cosmology(
+        Omega_c=0.27, Omega_b=0.045, h=0.67, sigma8=0.8, n_s=0.96,
+        transfer_function=model)
+    pk = ccl.Pk2D.pk_from_model(cosmo, model=model)
```

Comment:
So if I make a Cosmology object with BBKS but then call pk_from_model with model='eisenstein_hu', will the code fail? If it does, why do I need to pass the model string at all? This API confuses me a bit.

Reply:
No, this works for any model, regardless of what was chosen for the cosmology (i.e. you could use `transfer_function="boltzmann_class"` and then compute the BBKS power spectrum with this method. I just make the `transfer_function` flag coincide here so I can compare both power spectra further down. I can make this explicit if needed.

Reply:
I've made this explicitly clear.